src/map/mob.cpp

```diff
@@ -2416,6 +2416,81 @@ void mob_damage(struct mob_data *md, struct block_list *src, int damage)
 	}
 }
 
+/**
+ * Get modified drop rate
+ * @param src: Source object
+ * @param mob: Monster data
+ * @param base_rate: Base drop rate
+ * @param drop_modifier: RENEWAL_DROP level modifier
+ * @return Modified drop rate
+ */
+int mob_getdroprate(struct block_list *src, struct mob_db *mob, int base_rate, int drop_modifier)
+{
+	int drop_rate = base_rate;
+
+	if (battle_config.mob_size_influence) { // Change drops depending on monsters size [Valaris]
+		if (mob->status.size == SZ_MEDIUM && drop_rate >= 2)
+			drop_rate /= 2; // SZ_MEDIUM actually is small size modification... this is not a bug!
+		else if (mob->status.size == SZ_BIG)
+			drop_rate *= 2;
+	}
+
+	if (src) {
+		if (battle_config.drops_by_luk) // Drops affected by luk as a fixed increase [Valaris]
+			drop_rate += status_get_luk(src) * battle_config.drops_by_luk / 100;
+		if (battle_config.drops_by_luk2) // Drops affected by luk as a % increase [Skotlex]
+			drop_rate += (int)(0.5 + drop_rate * status_get_luk(src) * battle_config.drops_by_luk2 / 10000.);
+	}
+
+	if (src->type == BL_PC) { // Player specific drop rate adjustments
```

Comment:
This check block shld be added under if (src) to check whether the src has a value before doing any other checks

Reply:
What's the fix for this? just add nullptr?

Reply:
Done, moved the end of the if (src) block to the right position